### PR TITLE
Issue 3913 - Emit better lookup diagnostics for enum members

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Fixes https://issues.dlang.org/show_bug.cgi?id=3913

Recreated from https://github.com/D-Programming-Language/dmd/pull/3252

### CODE REVIEW

File: src/mtype.d [hunk 1]
```diff
@@ -8323,7 +8323,24 @@ public:
             {
                 return getProperty(e.loc, ident, flag);
             }
-            return sym.getMemtype(Loc()).dotExp(sc, e, ident, flag);
+
+            if (flag == 0)
+                flag = 1;  // silence lookup errors for memtype
+
+            Expression res = sym.getMemtype(Loc()).dotExp(sc, e, ident, flag);
```
Comment (yebblies) [inline src/mtype.d]: Just pass `1` instead of changing `flag`?

File: src/mtype.d [hunk 2]
```diff
@@ -8323,7 +8323,24 @@ public:
...
+            if (!res)
+            {
+                if (auto ns = sym.search_correct(ident))
+                    e.error("no property '%s' for type '%s'. Did you mean "
+                        "'%s.%s' ?", ident.toChars(), toChars(), toChars(),
```
Comment (yebblies) [inline src/mtype.d]: Probably best to keep the string literal on line line.

File: test/compilable/diag3913.d [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/diag3913.d(13): Error: no property 'foobardoo' for type 'Foo'
+fail_compilation/diag3913.d(14): Error: no property 'secon' for type 'Foo'. Did you mean 'Foo.second' ?
+---
+*/
+
+module diag3913;
```
Comment (yebblies) [inline test/compilable/diag3913.d]: Don't need a module declaration.

File: src/mtype.d (hunk 1 above)
Comment (9rnsr) [inline src/mtype.d]: And. if the original `flag` == 1, following improved diagnostic messages should not be displayed.

File: src/mtype.d (hunk 2 above)
Comment (AndrejMitrovic) [inline src/mtype.d]: Ah I guess we don't care about line length limits?

Comment (yebblies) [inline src/mtype.d]: It's not a hard limit, and it's not followed very often, and magicport removed all the manual line breaks.
Plus, implicit string literal concatenation is evil.

Comment (AndrejMitrovic) [inline src/mtype.d]: Nowadays I find `~` evil. :P

**Comment by AndrejMitrovic** [comment]
Updated with the requested changes. I don't get a segfault like the tester did previously, I'll try to see what's going on soon if it happens again.

**Comment by yebblies** [comment]
Line numbers need fixing

**Comment by AndrejMitrovic** [comment]
> Line numbers need fixing

Done.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**